lxd/networks.go

```diff
@@ -1334,6 +1334,47 @@ func networkLeasesGet(d *Daemon, r *http.Request) response.Response {
 
 	// Get all static leases.
 	if !isClusterNotification(r) {
+		// Get the downstream networks.
+		if instProjectName == project.Default && n.Managed && n.Type != "ovn" {
+			// Load all the networks.
+			var projectNetworks map[string]map[int64]api.Network
+			err = d.State().Cluster.Transaction(func(tx *db.ClusterTx) error {
+				projectNetworks, err = tx.GetCreatedNetworks()
+				return err
+			})
+			if err != nil {
+				return response.SmartError(err)
+			}
+
+			// Look for networks using the current network as an uplink.
+			for projectName, networks := range projectNetworks {
+				for _, network := range networks {
+					if network.Config["network"] != n.Name {
+						continue
+					}
+
+					// Found a network, add a lease.
+					ipv4 := network.Config["volatile.network.ipv4.address"]
```

Comment:
@stgraber please could you avoid the duplication for the different keys here by using a for loop, e.g.

```go
for _, key := range []string{"volatile.network.ipv4.address", "volatile.network.ipv6.address"} {
	ip := network.Config[key]
	if ip != "" {
		leases = append(leases, api.NetworkLease{
			Hostname: fmt.Sprintf("%s-%s.uplink", projectName, network.Name),
			Address:  ip,
			Type:     "uplink",
		})
	}
}
```

Reply:
done